infrastructure/base/main.tf

```diff
@@ -74,10 +49,85 @@ data "aws_subnets" "subnets_with_ec2_instance_type_offering_map" {
   }
 }
 
+module "iam" {
+  source = "./modules/iam"
+}
+
+resource "random_password" "api_token_salt" {
+  length           = 32
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "random_password" "admin_jwt_secret" {
+  length           = 32
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "random_password" "transfer_token_salt" {
+  length           = 32
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "random_password" "jwt_secret" {
+  length           = 32
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
 locals {
-  subnets_with_ec2_instance_type_offering_ids = sort([for k, v in data.aws_subnets.subnets_with_ec2_instance_type_offering_map : v.ids[0]])
+  subnets_with_ec2_instance_type_offering_ids = sort([
+    for k, v in data.aws_subnets.subnets_with_ec2_instance_type_offering_map : v.ids[0]
+  ])
+  staging_cms_env                             = {
+    HOST                = "0.0.0.0"
+    PORT                = 1337
+    APP_KEYS            = "toBeModified1,toBeModified2"
+    API_TOKEN_SALT      = random_password.api_token_salt.result
+    ADMIN_JWT_SECRET    = random_password.admin_jwt_secret.result
+    TRANSFER_TOKEN_SALT = random_password.transfer_token_salt.result
+    JWT_SECRET          = random_password.jwt_secret.result
+    CMS_URL             = "https://${var.staging_domain}/cms/"
+    NODE_ENV            = "development"
+
+    # Database
+    DATABASE_CLIENT                  = "postgres"
+    DATABASE_HOST                    = module.staging.postgresql_host
+    DATABASE_PORT                    = module.staging.postgresql_port
+    DATABASE_NAME                    = "afoco"
+    DATABASE_USERNAME                = module.staging.postgresql_username
+    DATABASE_PASSWORD                = module.staging.postgresql_password
+    DATABASE_SSL                     = true
+    DATABASE_SSL_REJECT_UNAUTHORIZED = false
+  }
+  staging_client_env = {
+    NEXT_PUBLIC_ENVIRONMENT      = development
```

Comment:
quotes here? or was that meant to be a var reference?

Reply:
definitely quotes here, and yet another example of crap that should not work and yet it does <3